Neuron/Models/Realm/TokenModel.swift

```diff
@@ -51,7 +49,7 @@ class TokenModel: Object, Decodable {
 
     var type: TokenType {
         if isNativeToken {
-            if chainId == NativeChainId.ethMainnetChainId {
+            if chain?.chainId == NativeChainId.ethMainnetChainId {
```

Comment:
Consider removing `ethMainnetChainId`. If Ethereum chain ID is not used we can just set `chain.chainId` for Ethereum to nil or empty string?

Reply:
YES, if it's Ethereum chain , the `chian?` will be nil.